spec/stylelint_spec.rb

```diff
@@ -12,35 +12,46 @@ module Danger
     # You should test your custom attributes and methods here
     #
     describe "with Dangerfile" do
-      before do
-        @dangerfile = testing_dangerfile
-        @my_plugin = @dangerfile.stylelint
+      let(:stylelint) { testing_dangerfile.stylelint }
 
-        # mock the PR data
-        # you can then use this, eg. github.pr_author, later in the spec
-        json = File.read("#{File.dirname(__FILE__)}/support/fixtures/github_pr.json") # example json: `curl https://api.github.com/repos/danger/danger-plugin-template/pulls/18 > github_pr.json`
-        allow(@my_plugin.github).to receive(:pr_json).and_return(json)
+      before do
+        allow(stylelint.git).to receive(:deleted_files).and_return([])
+        allow(stylelint.git).to receive(:added_files).and_return([])
+        allow(stylelint.git).to receive(:modified_files).and_return([])
+        allow(stylelint.git).to receive(:renamed_files).and_return([])
+        stub_const("Danger::DangerStylelint::DEFAULT_BIN_PATH", "spec/fixtures/stylelint.config.js")
       end
 
-      # Some examples for writing tests
-      # You should replace these with your own.
+      it "does not make an empty message" do
```

Comment:
I'm not sure what's this one for, is it to test the initial state of the report to make sure it's empty?

Reply:
This one is for the scenario when there's nothing to lint, danger won't comment anything!